otsdaq-mu2e/FEInterfaces/CFOFrontEndInterfaceImpl.cc

```diff
@@ -1561,28 +1561,24 @@ std::string CFOFrontEndInterface::SetRunplan(const std::string& binFilename)
 	__FE_COUT__ << "Set CFO Run Plan"  << __E__;
 
 	__FE_COUTV__(binFilename);
-	std::ifstream file(binFilename, std::ios::binary | std::ios::ate);
-	if (file.eof())
+
+	std::FILE* fp = std::fopen(binFilename.c_str(), "rb");
+	if(!fp)
 	{
-		__SS__ << ("Output File (" + binFilename + ") didn't open. Does it exist?") << __E__;
+		__SS__ << "Could not open file at " << binFilename << 
+			". Error: " << errno << " - " << strerror(errno) << __E__;
 		__SS_THROW__;
-		
 	}
-	mu2e_databuff_t inputData;
-	auto inputSize = file.tellg();
-	uint64_t dmaSize = static_cast<uint64_t>(inputSize) + 8;
-	file.seekg(0, std::ios::beg);
-	//*reinterpret_cast<uint64_t*>(inputData) = input.size();
-	memcpy(&inputData[0], &dmaSize, sizeof(uint64_t));
-	file.read(reinterpret_cast<char*>(&inputData[8]), inputSize);
-	file.close();
-	
+
+	std::string binaryContents;
+	std::fseek(fp, 0, SEEK_END);
+	binaryContents.resize(std::ftell(fp));
+	std::rewind(fp);
+	std::fread(&binaryContents[0], 1, binaryContents.size(), fp);
+	std::fclose(fp);
 
 	//set the file in hardware:
-	thisCFO_->GetDevice()->begin_dcs_transaction();
-	thisCFO_->GetDevice()->write_data(DTC_DMA_Engine_DCS, //CFO_DMA_Engine_RunPlan, 
-		inputData, sizeof(inputData));
-	thisCFO_->GetDevice()->end_dcs_transaction();
+	thisCFO_->SetRunPlanData(binaryContents, 0 /* address */);
```

Comment:
BRAM style CFO Run Plan is only verified in CFO version **CFO2024Aug01_13_22.1** and beyond

Reply:
CFO firmware branch is rrivera/rrivera/swappingCFOdmaChannelsBRAM in git@github.com:Mu2e/TDAQFirmware.git